src/kaocha/config.clj

```diff
@@ -217,6 +222,39 @@
                           tests)))))
     config))
 
+(defn load-config2
+  "Loads config from config-file, factoring in profile specified using profile,
+  and displaying messages about any errors."
+  ([config-file]
+   (load-config2 config-file nil {} nil nil))
+  ([config-file profile opts]
+   (load-config2 config-file profile opts nil nil))
+  ([config-file profile opts cli-options cli-args]
+   (let [config (cond-> config-file
+                    true (load-config-file (if profile (assoc opts :profile profile) opts))
+                    cli-options (apply-cli-opts cli-options)
+                    cli-args (apply-cli-args cli-args))
+
+         check_config_file (when (not (. (io/file (or config-file "tests.edn")) exists))
+                             (output/warn (format (str "Did not load a configuration file and using the defaults.\n"
+                                          "This is fine for experimenting, but for long-term use, we recommend creating a configuration file to avoid changes in behavior between releases.\n"
+                                          "To create a configuration file using the current defaults and configuration file location, create a file named %s that contains '#%s {}'.")
+                                                  config-file
+                                     current-reader)))
+         check  (try
+                  (specs/assert-spec :kaocha/config config)
+                  (catch AssertionError e
+                    (output/error "Invalid configuration file:\n"
+                                  (.getMessage e))
+                    (throw+ {:kaocha/early-exit 252})))]
+     (cond-> config
+       check_config_file (update ::warnings conj check_config_file)
+       check (update ::warnings conj check)))))
```

Comment:
Not sure I understand what this is meant to the doing. Both `check_config_file` and `check` will AFAICT return `nil` or throw an exception.